python/ray/tests/conftest.py

```diff
@@ -175,7 +175,9 @@ def start_redis(db_dir):
         for _ in range(redis_replicas()):
             # max port for redis cluster
             port = 55536
-            while port >= 55535:
+            # The port on macos needs to be greater than 49159,
+            # otherwise an error "bind: Address already in use" will be reported.
+            while port < 49159 or port >= 55535:
```

Comment:
Hmm I don't like this solution (what if s.bind doesn't resolve to the port between this number?). Is it possible to just try binding it from 49159 ~ 55535 instead?

Reply:
@rkooo567  No problem, I have made the changes, please review again.